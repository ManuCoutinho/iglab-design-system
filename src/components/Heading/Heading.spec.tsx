import { screen, render } from '@testing-library/react'
import { Heading } from '.'

describe('<Heading/>', () => {
  it('should render Heading correctly ', () => {
    render(<Heading>Text</Heading>)
    expect(screen.getByRole('heading', { name: /Text/i })).toBeInTheDocument()
  })
  it('should render Heading with right size', () => {
    render(<Heading size='lg'>Text</Heading>)
    expect(screen.getByRole('heading', { name: /Text/i })).toHaveClass(
      'text-2xl'
    )
  })
  it('should render Heading with className prop', () => {
    render(<Heading className='text-red-100'>Text</Heading>)
    expect(screen.getByRole('heading', { name: /Text/i })).toHaveClass(
      'text-red-100'
    )
  })
  it('match to snapshot', () => {
    const { container } = render(<Heading>Text</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
