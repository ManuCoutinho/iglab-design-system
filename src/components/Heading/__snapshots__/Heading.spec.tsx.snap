// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> match to snapshot 1`] = `
<h2
  class="text-gray-100 font-sans font-bold text-xl"
>
  Text
</h2>
`;
